embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/scheduling/SchedulingServiceImpl.kt

```diff
@@ -66,13 +63,9 @@ class SchedulingServiceImpl(
     private fun startDeliveryLoop() {
         // When a payload arrives, check to see if there's already an active job try to deliver payloads
         // If not, schedule job. If so, do nothing.
-        if (sendLoopActive.compareAndSet(false, true)) {
-            deliveryTracer?.onStartDeliveryLoop(true)
-            schedulingWorker.submit {
-                deliveryLoop()
-            }
-        } else {
-            deliveryTracer?.onStartDeliveryLoop(false)
+        deliveryTracer?.onStartDeliveryLoop()
+        schedulingWorker.submit {
+            deliveryLoop()
         }
     }
 
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line +78]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Change this method to return only the next payload to be delivered. It'll basically poll at every iteration and bail when there are none to be delivered.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/1743?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->